modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/WrapperWithQuantity.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+
+import QuantitySelector from '../quantity_selector/QuantitySelector';
+
+function WrapperWithQuantity({AddToCartButton, ...props}) {
+	const [currentQuantity, setCurrentQuantity] = useState();
+
+	const onQuantityUpdate = (quantity) => setCurrentQuantity(quantity);
```

Comment:
I don't think you need to re-declare a function here
```suggestion
	const onQuantityUpdate = setCurrentQuantity
```

You can probably just pass `setCurrentQuantity` directly at `onUpdate={setCurrentQuantity}`. So this intermediate variable isn't even necessary.

Reply:
Right, it's definitely neater and I thought of this too. I simply wanted to have an easily debuggable spot within this component instead of leaving everything implicit across the rendering/state changing flow. I'll fix it!